### PR TITLE
Fix typescript build errors, but perhaps with *implications*

### PR DESCRIPTION
**Pull request by thomasballinger**
I haven't tried to run this yet, just making the PR as a reminder to investigate

### CODE REVIEW

File: convex/tsconfig.json [hunk 1]
```diff
@@ -14,7 +14,7 @@
     "forceConsistentCasingInFileNames": true,
     "allowSyntheticDefaultImports": true,
     "module": "ESNext",
-    "moduleResolution": "Node",
+    "moduleResolution": "bundler",
```
Comment (ianmacartney) [inline convex/tsconfig.json:17]: Is this related to the `esbuild` config? fwiw, I got `convex-helpers` working only after using regular `tsc` and explicit exports for every module that uses an index.ts.

Comment (thomasballinger) [reply convex/tsconfig.json:17]: I don't think so, code should work at runtime with both settings. This is just a TypeScript thing, the openai types seem broken for a particular type import when using old Node module resolution.

**Comment by xixixao** [comment]
Fixed on main in 5c7a35e, thanks!
